.github/workflows/tests.yml

```diff
@@ -60,3 +60,162 @@ jobs:
           flags: tests
           env_vars: PYTHON
           name: ${{ matrix.python }}
+
+  cygwin:
+    runs-on: windows-latest
+    env:
+      FORCE_COLOR: true
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - '3.9'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Cygwin
+        uses: cygwin/cygwin-install-action@v2
+        with:
+          packages: >-
+            python39
+            python39-devel
+            python39-pip
```

Comment:
I put the Python version in the matrix mostly as a reminder of which Python version this job runs. I don't expect Cygwin to update the Python version they distribute anytime soon.

Reply:
The other option would be to try testing on python 3.7 or 3.8, also currently available from Cygwin, and the reason the package is no longer named python3 and python3-pip.  I don't think that would turn up too many more bug, though.

Reply:
The Cygwin tests are interesting because of the peculiarities of the platform. I don't expect differences between Python versions. We already have many CI jobs and I'm more thinking or remove some elements of the test matrix rather than adding some more.